worker/src/services/IngestionService/index.ts

```diff
@@ -590,24 +622,93 @@ export class IngestionService {
     };
     const { projectId, entityId, table } = params;
 
-    return await instrumentAsync({ name: `get-${table}` }, async () => {
-      const queryResult = await this.clickhouseClient.query({
-        query: `SELECT * FROM ${table} WHERE project_id = '${projectId}' AND id = '${entityId}' ORDER BY event_ts DESC LIMIT 1 by id, project_id SETTINGS use_query_cache = false;`,
-        format: "JSONEachRow",
-      });
+    return await instrumentAsync(
+      { name: `get-clickhouse-${table}` },
+      async () => {
+        const queryResult = await this.clickhouseClient.query({
+          query: `
+            SELECT *
+            FROM ${table}
+            FINAL
+            WHERE project_id = {projectId: String}
+            AND id = {entityId: String}
+            ORDER BY event_ts DESC
+            LIMIT 1 BY id, project_id SETTINGS use_query_cache = false;
+          `,
+          format: "JSONEachRow",
+          query_params: { projectId, entityId },
+        });
 
-      const result = await queryResult.json();
+        const result = await queryResult.json();
 
-      if (result.length === 0) return null;
+        if (result.length === 0) return null;
 
-      return table === TableName.Traces
-        ? convertTraceReadToInsert(recordParser[table].parse(result[0]))
-        : table === TableName.Scores
-          ? convertScoreReadToInsert(recordParser[table].parse(result[0]))
-          : convertObservationReadToInsert(
-              recordParser[table].parse(result[0]),
-            );
-    });
+        return table === TableName.Traces
+          ? convertTraceReadToInsert(recordParser[table].parse(result[0]))
+          : table === TableName.Scores
+            ? convertScoreReadToInsert(recordParser[table].parse(result[0]))
+            : convertObservationReadToInsert(
+                recordParser[table].parse(result[0]),
+              );
+      },
+    );
+  }
+
+  private async getPostgresRecord(params: {
+    projectId: string;
+    entityId: string;
+    table: TableName.Traces;
+  }): Promise<TraceRecordInsertType | null>;
+  private async getPostgresRecord(params: {
+    projectId: string;
+    entityId: string;
+    table: TableName.Scores;
+  }): Promise<ScoreRecordInsertType | null>;
+  private async getPostgresRecord(params: {
+    projectId: string;
+    entityId: string;
+    table: TableName.Observations;
+  }): Promise<ObservationRecordInsertType | null>;
+  private async getPostgresRecord(params: {
+    projectId: string;
+    entityId: string;
+    table: TableName;
+  }) {
+    const recordParser = {
+      traces: convertPostgresTraceToInsert,
+      scores: convertPostgresScoreToInsert,
+      observations: convertPostgresObservationToInsert,
+    };
+    const { projectId, entityId, table } = params;
+
+    return await instrumentAsync(
+      { name: `get-postgres-${table}` },
```

Comment:
nit: we can remove this as prisma has instrumentation built in.

Reply:
Addressed in https://github.com/langfuse/langfuse/pull/3978/commits/4a99a1008368155c3f56844267eb84c9a41c4748